services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -266,6 +270,48 @@ async def get_file(
         ) from err
 
 
+@router.get(
+    ":search",
+    response_model=Page[File],
+    responses={**_COMMON_ERROR_RESPONSES},
+)
+async def search_files_page(
+    storage_client: Annotated[StorageApi, Depends(get_api_client(StorageApi))],
+    user_id: Annotated[int, Depends(get_current_user_id)],
+    page_params: Annotated[PaginationParams, Depends()],
+    sha256_checksum: SHA256Str | None = None,
+    file_id: UUID | None = None,
+):
+    """Search files"""
+    # TODO currently this method builds the page itself. That should be implemented in storage
```

Comment:
if you do not plan on doing this nowish. Please create an issue and convert this TODO to a NOTE, with the reference to the issue.
Also don't forget to add the t:maintenance label on the issue.

Reply:
Done